pkg/codegen/internal/test/testdata/aws-fargate.pp.go

```diff
@@ -166,7 +166,9 @@ func main() {
 					ContainerPort:  pulumi.Int(80),
 				},
 			},
-		})
+		}, pulumi.DependsOn([]string{
```

Comment:
This should be `[]pulumi.Resource`

Reply:
That's right, yes. I was surprised to see this defaulting to `[]string` instead of `[]interface{}` given that I was passing in `list(dynamic)`.

Reply:
Yeah, just looks like a bug. Try this patch:

```diff
diff --git a/pkg/codegen/go/gen_program_expressions.go b/pkg/codegen/go/gen_program_expressions.go
index a3a61c9b..486d6fe4 100644
--- a/pkg/codegen/go/gen_program_expressions.go
+++ b/pkg/codegen/go/gen_program_expressions.go
@@ -622,7 +622,7 @@ func (g *generator) argumentTypeName(expr model.Expression, destType model.Type,
 				return "pulumi.Float64"
 			}
 			return "float64"
-		case model.StringType, model.DynamicType:
+		case model.StringType:
 			if isInput {
 				return "pulumi.String"
 			}
@@ -632,6 +632,11 @@ func (g *generator) argumentTypeName(expr model.Expression, destType model.Type,
 				return "pulumi.Bool"
 			}
 			return "bool"
+		case model.DynamicType:
+			if isInput {
+				return "pulumi.Any"
+			}
+			return "interface{}"
 		default:
 			return destType.Name
 		}

```

Reply:
No worries--we can take care of that in a follow-up change.